src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -195,6 +196,111 @@ export class FileService implements IFileService {
+		const ret = new EventEmitter();
+
+		fs.open(resource.path, 'r', (err, fd) => {
+
+			if (err) {
+				ret.emit('error', err);
+				return;
+			}
+
+			let decoder: NodeJS.ReadWriteStream;
+			let totalBytesRead = 0;
+
+			const finish = (err?: any) => {
+				if (err) {
+					ret.emit('error', err);
+				}
+				fs.close(fd, err => {
+					if (err) {
+						ret.emit('error', err);
+					}
+					if (decoder) {
+						decoder.removeAllListeners();
+					}
+					ret.emit('end');
+				});
+			};
+
+			const handleChunk = (bytesRead) => {
+				if (token.isCancellationRequested) {
+					// cancellation
+					finish(new Error('cancelled'));
+
+				} else if (bytesRead < FileService.chunkSize) {
+					// done, write rest, end
+					decoder.write(FileService.chunkBuffer.slice(0, bytesRead), finish);
+
+				} else {
+					// read, write, repeat
+					decoder.write(FileService.chunkBuffer, readChunk);
+				}
+			};
+
+			const readChunk = () => {
+				fs.read(fd, FileService.chunkBuffer, 0, FileService.chunkSize, null, (err, bytesRead) => {
+					totalBytesRead += bytesRead;
+
+					if (totalBytesRead > MAX_FILE_SIZE) {
+						// reading too much... this looks stupid because we have already read much
+						// but it safes an upfront stat call for every 'normal' file we open.
+						finish(new FileOperationError(
+							nls.localize('fileTooLargeError', "File too large to open"),
+							FileOperationResult.FILE_TOO_LARGE
+						));
+					} else if (err) {
+						// some error happened
+						finish(err);
+
+					} else if (decoder) {
+						//
+						handleChunk(bytesRead);
+
+					} else {
+						// when receiving the first chunk of data we need to create the
+						// decoding stream which is then used to drive the string stream.
+						Promise.resolve(detectMimeAndEncodingFromBuffer(
+							{ buffer: FileService.chunkBuffer, bytesRead },
+							options && options.autoGuessEncoding)
+						).then(value => {
+
+							if (options && options.acceptTextOnly && value.mimes.indexOf(baseMime.MIME_BINARY) >= 0) {
+								// Return error early if client only accepts text and this is not text
+								finish(new FileOperationError(
+									nls.localize('fileBinaryError', "File seems to be binary and cannot be opened as text"),
+									FileOperationResult.FILE_IS_BINARY
+								));
+
+							} else {
+								value.encoding = this.getEncoding(resource, this.getEncoding2(resource, options, value));
+								ret.emit('encoding', value.encoding);
+								decoder = encoding.decodeStream(value.encoding);
+								decoder.on('data', arg => ret.emit('data', arg));
+								decoder.on('error', arg => ret.emit('error', arg));
+								handleChunk(bytesRead);
+							}
+
+						}).catch(err => {
+							// failed to get encoding
+							finish(err);
+						});
+					}
+				});
+			};
+
+			readChunk();
+		});
+
+		return ret;
+	}
+
+	//#endregion
+
 	public resolveStreamContent(resource: uri, options?: IResolveContentOptions): TPromise<IStreamContent> {
```

Comment:
This method is now not used anymore, I suggest to keep the interface and fold your work in `resolveStringStream` into this method.